File: Assets/scenes/hyperspace/scripts/CometEvent.js
#pragma strict


var gravityFailSfx : AudioClip;	//sound to play during failed exit

private var theShip : GameObject; //the ship
private var running : boolean = false;

private var startTime : float = 0;
public var triggerTime : float = 11;
public var cometPrefab : Transform;

function Start () {
	if(theShip == null){
		theShip = GameObject.Find("TheShip");
	}
	cometPrefab.parent = null;
	startTime = Time.fixedTime;
}

function startSequence(){
	
	
	
	running = true;
	
	//start moving the asteroid
	cometPrefab.GetComponent.<CometBehaviour>().velocity = Vector3(0,-24.9,0);
	yield WaitForSeconds(8f);
	cometPrefab.GetComponent.<CometBehaviour>().velocity = Vector3(0,0,0);
	//yield WaitForSeconds(2f);
	AudioSource.PlayClipAtPoint(gravityFailSfx, transform.position);
}

function FixedUpdate () {
	if(running){
		
		
	}
	if(Time.fixedTime - startTime > triggerTime && !running){
		startSequence();
	}
}

File: Assets/scenes/hyperspace/scripts/CometEvent.cs
using UnityEngine;
using System;
using System.Collections;


public class CometEvent:HyperSpaceEvent{
	
	public AudioClip gravityFailSfx;	//sound to play during failed exit
	
	GameObject theShip; //the ship
	bool running = false;
	
	float startTime = 0.0f;
	//public float triggerTime = 11.0f;
	public Transform cometPrefab;

	Vector3 cometVel = Vector3.zero;

	
	public void Start() {
		if(theShip == null){
			theShip = GameObject.Find("TheShip");
		}
		cometPrefab.parent = null;
		startTime = Time.fixedTime;
		triggerDelay = 8.0f;
	}
	
	public override IEnumerator startSequence(){
		if(running == false){
			
			
			running = true;
			
			//start moving the asteroid
			cometVel = new Vector3(0.0f,-24.9f,0.0f);
			yield return new WaitForSeconds(8f);
			cometVel = new Vector3(0.0f,0.0f,0.0f);
			//yield WaitForSeconds(2f);
			UsefulShit.PlayClipAt(gravityFailSfx, transform.position);
		}
	}
	
	public void FixedUpdate() {
		if(running){
			cometPrefab.transform.position += cometPrefab.TransformDirection(cometVel);
			
		}

	}
}
